embedded/store/immustore.go

```diff
@@ -2550,30 +2552,31 @@ func (s *ImmuStore) ReadValue(entry *TxEntry) ([]byte, error) {
 		return nil, ErrExpiredEntry
 	}
 
+	if entry.vLen == 0 {
+		return nil, nil
+	}
+
 	b := make([]byte, entry.vLen)
 
-	bval, err := s.readValueAt(b, entry.vOff, entry.hVal)
+	_, err := s.readValueAt(b, entry.vOff, entry.hVal, false)
 	if err != nil {
 		return nil, err
 	}
 
-	// if bval == 0, the value is nil
```

Comment:
should not nil be returned if data is nil? We are still returning an array of n bytes which are 0s when data is empty

Reply:
it should be equivalent, when committing a transaction, either nil or empty values are treated the same way. So it should be equivalent when returning the value. Checking the length of the returned value unifies them.

Nevertheless, I though it was done in this PR otherwise it's on the bidirectional streaming one, I have this change already
```
if entry.vLen == 0 {
		return nil, nil
	}
```

So `nil` is returned even before allocating the buffer or reading data.

Reply:
just checked, that change is already in this PR at line 2555